core/src/openid.cpp

```diff
@@ -241,18 +237,31 @@ namespace irods::http::openid
 		// Use a realm_secret if provided. It should be base64url encoded, as the key might
 		// not be ASCII printable.
 		if (auto realm_secret{irods::http::globals::oidc_configuration().find("realm_secret")};
-		    realm_secret != std::end(irods::http::globals::oidc_configuration()))
+		    _type == TokenType::Access && realm_secret != std::end(irods::http::globals::oidc_configuration()))
 		{
 			key = jwt::base::decode<jwt::alphabet::base64url>(
 				jwt::base::pad<jwt::alphabet::base64url>(realm_secret->get_ref<const std::string&>()));
 		}
+		// Some OpenID Providers do not follow OpenID Connect Core 1.0 incorporating errata set 2,
+		// Section 3.1.3.7, Bullet Point 8. This variable would allow for non-standard behavior, if provided.
+		else if (auto id_token_secret{irods::http::globals::oidc_configuration().find("nonstandard_id_token_secret")};
+		         _type == TokenType::ID &&
+		         id_token_secret != std::end(irods::http::globals::oidc_configuration()))
+		{
+			key = jwt::base::decode<jwt::alphabet::base64url>(
+				jwt::base::pad<jwt::alphabet::base64url>(id_token_secret->get_ref<const std::string&>()));
+		}
```

Comment:
Should we update the comment above on L233-234 to reflect this third option (currently only mentions "realm_secret" or "client_secret")?

Reply:
Updated.